src/lua/LuaFileSystem.h

```diff
@@ -4,13 +4,16 @@
 #ifndef _LUAFILESYSTEM_H
 #define _LUAFILESYSTEM_H
 
+#include <string>
+
+struct lua_State;
+typedef int (*lua_CFunction) (lua_State* L);
+
```

Comment:
There's not really a need to forward-declare here; this header isn't included anywhere which isn't also including the main Lua headers. By convention we include the "main" `lua/Lua.h` header to get access to Lua typedefs.

```suggestion
#include "lua/Lua.h"

```